localdev/docker-compose.yaml

```diff
@@ -0,0 +1,131 @@
+version: "3.8"
+
+services:
+  #
+  # Infrastructure services needed for Armada
+  # Note that kubernetes cluster is also needed (kind) but is external to the compose environment
+  #
+  redis:
+    container_name: redis
+    image: redis
+    ports:
+      - "6379:6379"
+
+  postgres:
+    container_name: postgres
+    image: postgres
+    environment:
+      - "POSTGRES_PASSWORD=psw"
+    ports:
+      - "5432:5432"
+
+  pulsar:
+    image: kezhenxu94/pulsar
```

Comment:
What do you think about going ahead and using profiles for each of these - have a `profiles: ["macos"]` and `profiles: ["linux"]` block? People will have to get used to specifying, e.g. `--profile macos` when invoking `docker compose`, but shell command history will make that pretty easy after a while.

Reply:
ok I'll try it -- the run.sh script can default the right one for you, too

Reply:
well profiles get a little messy -- we'd need "downstream" profile config for things that depend_on pulsar -- so alternative armada-servers, etc. Trying an env-var that defaults to linux but gets the arm64 image name in run.sh if needed.